File: Hash Table/2441_largest_positive_integer_that_exists_with_its_negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        hashmap = {}
        max_ele = -1

        for num in nums:
            if -num in hashmap and abs(num) > max_ele:
                max_ele = abs(num)
            hashmap[num] = 1
            
        return max_ele

File: Array/2441_largest_positive_integer_that _exists_with_its_negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        unordered_map<int, int> hashmap;
        int res = -1;

        for(auto num: nums) {
            if(hashmap.find(-num) == hashmap.end()) hashmap[num] = 1;
            else res = max(res, abs(num));
        }

        return res;
    }
};